specification/databricks/resource-manager/Microsoft.Databricks/stable/2024-05-01/databricks.json

```diff
@@ -0,0 +1,2016 @@
+        "publicNetworkAccess": {
+          "type": "string",
+          "description": "The network access type for accessing workspace. Set value to disabled to access workspace only via private link.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "PublicNetworkAccess"
+          }
+        },
+        "requiredNsgRules": {
+          "type": "string",
+          "description": "Gets or sets a value indicating whether data plane (clusters) to control plane communication happen over private endpoint. Supported values are 'AllRules' and 'NoAzureDatabricksRules'. 'NoAzureServiceRules' value is for internal use only.",
+          "enum": [
+            "AllRules",
+            "NoAzureDatabricksRules",
+            "NoAzureServiceRules"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "RequiredNsgRules"
+          }
+        },
+        "defaultCatalog": {
+          "description": "Properties for Default Catalog configuration during workspace creation.",
+          "$ref": "#/definitions/DefaultCatalogProperties"
+        },
+        "isUcEnabled": {
+          "readOnly": true,
+          "description": "Indicates whether unity catalog enabled for the workspace or not.",
+          "type": "boolean"
+        },
+        "accessConnector": {
+          "description": "Access Connector Resource that is going to be associated with Databricks Workspace",
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "format": "arm-id",
+              "description": "The resource ID of Azure Databricks Access Connector Resource.",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.Databricks/accessConnectors"
+                  }
+                ]
+              }
+            },
+            "identityType": {
+              "type": "string",
+              "description": "The identity type of the Access Connector Resource.",
+              "enum": [
+                "SystemAssigned",
+                "UserAssigned"
+              ],
+              "x-ms-enum": {
+                "modelAsString": true,
+                "name": "IdentityType"
+              }
+            },
+            "userAssignedIdentityId": {
+              "type": "string",
+              "format": "arm-id",
+              "description": "The resource ID of the User Assigned Identity associated with the Access Connector Resource. This is required for type 'UserAssigned' and not valid for type 'SystemAssigned'.",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.ManagedIdentity/userAssignedIdentities"
+                  }
+                ]
+              }
+            }
```

Comment:
Please check whether its refactorable - is it possible to reference any of the existing common Definitions of resource 'identity' from common-types/v5 instead of redefining?

Reply:
No this is not common type, common type supports identity where identity type can be SA, UA as well and userAssignedIdentities is a dictionary here it is only a string. The use case is different.

Reply:
How is the use case different? To me, the use case sounds the same, just with some added restrictions, that shouldn't necessarily change the shape of the API.

Reply:
Sounding same is not going to make it same :), Like I said earlier resource level Identity object has two fileds 

IdentityType and UserAssignedIdentities dictionary. The use case there would be any resource can have multiple MIs, but the use case here is our workspace is using an MI of a resource called "accessConnector" and then performs the necessary MI operations. So this is different.